WhisperAPI/WhisperAPI/Models/ConversationContext.cs

```diff
@@ -25,5 +28,11 @@ public ConversationContext()
         public List<SearchQuery> SearchQueries { get; set; }
 
         public HashSet<SuggestedDocument> SuggestedDocuments { get; set; }
+
+        public HashSet<Question> Questions { get; set; }
```

Comment:
The class Quesiton does not implement GetHashCode and Equals. This means that the HashSet doesn't know how to hash and compare Quesiton, which defaults to reference equality. In other words, all questions will be different unless it is the very same object treference, which is not what we want. Do not implement those methods by hand, you can use VS or Resharper to generate the hashcode and equals override for you in the Question class. Place your cursor on Quesiton in public class Quesiton. Then, press Ctrl + . (period). You will see "Generate hashcode and equals" or something in the list. Then check the property "Id", we don't care about the text, all we care about is the Id. In other words, the hash must be only calculated based on the Id, not the string and same goes with equality check, because the Id is unique anyway.